kernel/src/fs/pipe.rs

```diff
@@ -32,43 +32,51 @@ const PIPE_BUF: usize = 4096;
 #[cfg(ktest)]
 const PIPE_BUF: usize = 2;
 
-pub fn new_pair() -> Result<(Arc<PipeReader>, Arc<PipeWriter>)> {
+/// Creates a pair of connected pipe file handles with the default capacity.
+pub fn new_pair_file() -> Result<(Arc<PipeReaderFile>, Arc<PipeWriterFile>)> {
+    new_pair_file_with_capacity(DEFAULT_PIPE_BUF_SIZE)
+}
+
+pub(super) fn new_pair() -> (PipeReader, PipeWriter) {
     new_pair_with_capacity(DEFAULT_PIPE_BUF_SIZE)
 }
 
-fn new_pair_with_capacity(capacity: usize) -> Result<(Arc<PipeReader>, Arc<PipeWriter>)> {
+fn new_pair_file_with_capacity(
+    capacity: usize,
+) -> Result<(Arc<PipeReaderFile>, Arc<PipeWriterFile>)> {
+    let (reader, writer) = new_pair_with_capacity(capacity);
+
+    Ok((
+        PipeReaderFile::new(reader, StatusFlags::empty())?,
+        PipeWriterFile::new(writer, StatusFlags::empty())?,
+    ))
+}
+
+fn new_pair_with_capacity(capacity: usize) -> (PipeReader, PipeWriter) {
     let (producer, consumer) = RingBuffer::new(capacity).split();
     let (producer_state, consumer_state) =
         Endpoint::new_pair(EndpointState::default(), EndpointState::default());
 
-    Ok((
-        PipeReader::new(consumer, consumer_state, StatusFlags::empty())?,
-        PipeWriter::new(producer, producer_state, StatusFlags::empty())?,
-    ))
+    (
+        PipeReader::new(consumer, consumer_state),
+        PipeWriter::new(producer, producer_state),
+    )
 }
 
-pub struct PipeReader {
+pub(super) struct PipeReader {
     consumer: Mutex<RbConsumer<u8>>,
-    state: Endpoint<EndpointState>,
-    status_flags: AtomicU32,
+    pub(super) state: Endpoint<EndpointState>,
```

Comment:
Can we still consider storing status_flags in `PipeReader` and `PipeWriter` but adjusting the NamedPipe implementation? When opening a `NamedPipe`, we can store the status flags to the underlying pipe reader/pipe writer. The benefit is that we can leave the pipe related code unchanged. And we don't need to add status flags as parameter for `FileIo::read/FileIo::write`. But we need to add APIs like `status_flags` and `set_status_flags` to `FileIo`.